### DISCUSSION TITLE
How do i start digging into this and tweaking it?

### DISCUSSION THREAD

**Discussion by immjs** [coder/code-server#2140] [Q&A]
Sorry for this pretty vague question but i need to tweak it a little to make it work in my context :
```text
Plugin (Has data) => Client => Server
   \==============================^
```
So basically i need a server that has the code-server to be able to share files with the Plugin with permission to edit and send commands and i wanted to know how I would approach this problem
I think using code-server would be neat, tho if you have an approach that would benefit my project more please inform me

I am also looking for a way to make this without a server, like having the plugin run ther server instead

Sorry for all the stuff i'm appending in here, It's just that it's been 3 days since i've tried to figure out a solution

**Comment by exvuma** [comment]
Sounds like an interesting project. Could you explain what plugin and commands you are referring to?

**Reply by immjs** [reply]
So my project is, to let the user install a plugin so that he can edit some heroku project in the browser and to add + commit + push every time a chabge is seen

**Reply by immjs** [reply]
Tho the user doesn't forcibly have node/npm so i won't be able to let shelljs run npm i codeserver

**Reply by exvuma** [reply]
thanks for explaining that. while plugging different apps into code-server may be added in the future, right now the easiest way to do this is probably to use an extension for git and set up your git provider to build Heroku on pushes
